src/screens/PinScreen.js

```diff
@@ -157,6 +159,10 @@ class PinScreen extends React.Component {
       if (cb) {
         cb(pin);
       }
+
+      if (this.props.route.params.dispatchResponse) {
+        pinScreenResult(pin);
+      }
```

Comment:
I could have passed a callback in the route params, but I don't think that's ideal as it yields a warning, I created this new param `dispatchResponse` as a new mechanism for getting the result in the sagas

We should refactor this in a future PR to stop passing callbacks as params